packages/zoltan2/example/block/kokkosBlock.cpp

```diff
@@ -64,11 +62,10 @@ using Teuchos::RCP;
  *  \todo write some examples that don't use teuchos
  */
 
-int main(int argc, char *argv[]) {
-  Teuchos::GlobalMPISession mpiSession(&argc, &argv);
-  Kokkos::initialize(argc, argv);
+int main(int narg, char *arg[]) {
+  Tpetra::ScopeGuard tscope(&narg, &arg);
+  Teuchos::RCP<const Teuchos::Comm<int> > comm = Tpetra::getDefaultComm();
```

Comment:
It looks like the corresponding `Kokkos::finalize` is still at the bottom of the file.  ScopeGuard will be grumpy about the double finalize.  Unfortunately, Kokkos only has an `is_initialized()` function, not an `is_finalized()` function.  (That's different than MPI, alas.)  This means that Tpetra can't check whether the user made a mistake like this.